File: src/main/java/com/indra87g/utils/MessageHandler.java
package com.indra87g.utils;

import cn.nukkit.command.CommandSender;
import cn.nukkit.utils.Config;
import cn.nukkit.utils.TextFormat;
import com.indra87g.Main;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

public class MessageHandler {

    private static Config messageConfig;
    private static final Map<String, String> messages = new HashMap<>();
    private static String prefix = "";

    public static void loadMessages(Main plugin) {
        File messagesFile = new File(plugin.getDataFolder(), "messages.yml");
        if (!messagesFile.exists()) {
            plugin.saveResource("messages.yml", false);
        }
        messageConfig = new Config(messagesFile, Config.YAML);
        prefix = TextFormat.colorize(messageConfig.getString("prefix", "&7[&eWaffle&7] &r"));

        for (String key : messageConfig.getKeys(false)) {
            if (!key.equals("prefix")) {
                messages.put(key, messageConfig.getString(key));
            }
        }
    }

    public static void sendMessage(CommandSender sender, String key, String... replacements) {
        String message = messages.getOrDefault(key, "&cMessage not found: " + key);
        message = replacePlaceholders(message, replacements);
        sender.sendMessage(prefix + TextFormat.colorize(message));
    }

    public static void sendDirectMessage(CommandSender sender, String message) {
        sender.sendMessage(TextFormat.colorize(message));
    }

    public static String getMessage(String key) {
        return messages.getOrDefault(key, "&cMessage not found: " + key);
    }

    public static String getMessage(String key, String... replacements) {
        String message = getMessage(key);
        return replacePlaceholders(message, replacements);
    }

    private static String replacePlaceholders(String message, String... replacements) {
        if (replacements.length % 2 != 0) {
            return message; // Invalid replacements
        }
        for (int i = 0; i < replacements.length; i += 2) {
            message = message.replace(replacements[i], replacements[i + 1]);
        }
        return message;
    }
}

File: src/main/kotlin/com/indra87g/utils/MessageHandler.kt
package com.indra87g.utils

import cn.nukkit.command.CommandSender
import cn.nukkit.utils.Config
import cn.nukkit.utils.TextFormat
import com.indra87g.Main
import java.io.File

object MessageHandler {

    private val messages = mutableMapOf<String, String>()
    private var prefix = ""

    fun loadMessages(plugin: Main) {
        val messagesFile = File(plugin.dataFolder, "messages.yml")
        if (!messagesFile.exists()) {
            plugin.saveResource("messages.yml", false)
        }
        val messageConfig = Config(messagesFile, Config.YAML)
        prefix = TextFormat.colorize(messageConfig.getString("prefix", "&7[&eWaffle&7] &r"))

        for (key in messageConfig.keys) {
            if (key != "prefix") {
                messages[key] = messageConfig.getString(key)
            }
        }
    }

    fun sendMessage(sender: CommandSender, key: String, vararg replacements: String) {
        var message = messages.getOrDefault(key, "&cMessage not found: $key")
        message = replacePlaceholders(message, *replacements)
        sender.sendMessage(prefix + TextFormat.colorize(message))
    }

    fun sendDirectMessage(sender: CommandSender, message: String) {
        sender.sendMessage(TextFormat.colorize(message))
    }

    fun getMessage(key: String): String {
        return messages.getOrDefault(key, "&cMessage not found: $key")
    }

    fun getMessage(key: String, vararg replacements: String): String {
        var message = getMessage(key)
        return replacePlaceholders(message, *replacements)
    }

    private fun replacePlaceholders(message: String, vararg replacements: String): String {
        if (replacements.size % 2 != 0) {
            return message // Invalid replacements
        }
        var tempMessage = message
        for (i in replacements.indices step 2) {
            tempMessage = tempMessage.replace(replacements[i], replacements[i + 1])
        }
        return tempMessage
    }
}
